src/validator/task.rs

```diff
@@ -796,16 +796,34 @@ fn connect_block(
             }
             None => (),
         };
-        if handle_m8(
+        match handle_m8(
             transaction,
             &accepted_bmm_requests,
             &prev_mainchain_block_hash,
-        )? {
-            tracing::trace!(
-                "Handled valid M8 BMM request in tx `{}`",
-                transaction.compute_txid()
-            );
-        }
+        ) {
+            Ok(true) => {
+                tracing::trace!(
+                    "Handled valid M8 BMM request in tx `{}`",
+                    transaction.compute_txid()
+                );
+            }
+            // This was not a BMM request
+            Ok(_) => (),
+
+            // Make sure to NOT propagate this outwards. These errors are not
```

Comment:
If we want to distinguish between fatal and nonfatal errors, we should do so using something like https://docs.rs/fatality/latest/fatality/index.html